packages/loopring_v3/contracts/aux/bridge/Bridge.sol

```diff
@@ -160,8 +168,22 @@ contract Bridge is IBridge, BatchDepositor, Claimable
         ctx.tokensOffset = tokensOffset;
         ctx.tokens = tokens;
 
+        // Process potential L1->L2 batch deposits if there are any.
+        // Batch deposits are not always necessary, if users have
+        // tokens on L2 already, this step is totally optional.
         _processDepositBatches(ctx, batches);
-        _processConnectorCalls(ctx, calls);
+
+        // Process L2 transfers from users to the Bridge, then withdrawl tokens
+        // to L1 to interact with connectors,
+        uint[]                      memory totalAmounts;
+        IBatchDepositor.Deposit[][] memory depositsList;
+        (totalAmounts, depositsList) = _processConnectorCalls(ctx, calls);
+
+        // Verify withdrawals
+        _verifyWithdrawals(ctx, totalAmounts);
+
+        // Do all resulting transfers back from the bridge to the users
+        _batchDeposit(address(this), depositsList);
```

Comment:
Any reason not to do this in a separate `_processConnectorCalls` call like before? `_processDepositBatches` and `_processConnectorCalls` or 2 completely separate operations and these don't have anything to do with each other so it makes sense to me to give each its own function and inputs.

Reply:
I think `_processDepositBatches` should be moved to `BatchDepositor`.

Reply:
> _processDepositBatches

If we do this, then _readTransfers, and Context structure will also be moved to the depositor.

Reply:
> Any reason not to do this in a separate `_processConnectorCalls` call like before? `_processDepositBatches` and `_processConnectorCalls` or 2 completely separate operations and these don't have anything to do with each other so it makes sense to me to give each its own function and inputs.

I don't feel strongly about my change, but I think it doesn't hurt to split the main logic inside the _processTransactions function to lay out the major steps. If there a performance impact, I will revert it back (are there?)

Reply:
> f we do this, then _readTransfers, and Context structure will also be moved to the depositor. But the Context is more Bridge specific, so I think we should keep it inside the Bridge code.

Oh, I see. It's a bit of a strange now that batch deposits request are made in contract A and they are processed in contract B, so the split doesn't really work 100% like it's currently implemented but that's fine.

> I don't feel strongly about my change, but I think it doesn't hurt to split the main logic inside the _processTransactions function to lay out the major steps. If there a performance impact, or if you prefer, I will revert it back (are there?)

I think I just see the split a bit differently then you. For me all batch deposit stuff is fully separate from all the bridge transactions stuff, so each is handled in it's own function, `_processTransactions` just makes sure the data they need is given correctly. It seems you see it more like one thing where batched deposits are an optional part of the complete flow. I don't really see it that way because the batched deposits that are processed in `_processTransactions` are unrelated to the bridge transactions. They could be processed in completely separate contracts (which is why the split makes sense), the only reason not to do so is performance.

It's fine for me like this, I don't think it has a performance impact but I'll have to measure it to be sure.

Reply:
Brecht, feel free to create a PR for improving the split. For me, the split makes it easier to understand the code, the original Bridge.sol is too big. The BatchDepositor is not a complete feature by itself, that's why it is now an abstract contract.